plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/TestHiveMetastoreCatalogs.java

```diff
@@ -0,0 +1,170 @@
+                .put("hive.metastore.thrift.catalog-name", "hive") // HiveMetastore uses "hive" as the default catalog name
+                .put("fs.hadoop.enabled", "false")
+                .put("fs.native-s3.enabled", "true")
+                .put("s3.path-style-access", "true")
+                .put("s3.region", MINIO_REGION)
+                .put("s3.endpoint", "http://" + hiveMinioDataLake.getMinio().getMinioApiEndpoint())
+                .put("s3.aws-access-key", MINIO_ACCESS_KEY)
+                .put("s3.aws-secret-key", MINIO_SECRET_KEY)
+                .buildOrThrow());
+
+        return queryRunner;
+    }
+
+    @Test
+    public void testShowTables()
+    {
+        assertThat(query("SHOW SCHEMAS")).matches("VALUES VARCHAR 'default', VARCHAR 'information_schema'");
+
+        HiveMetastore metastore = getConnectorService(getQueryRunner(), HiveMetastoreFactory.class)
+                .createMetastore(Optional.empty());
+
+        String defaultCatalogSchema = "default_catalog_schema";
+        metastore.createDatabase(createDatabaseMetastoreObject(defaultCatalogSchema, Optional.of("s3://%s/%s".formatted(bucketName, defaultCatalogSchema))));
+
+        Session defaultCatalogSession = testSessionBuilder()
+                .setCatalog("hive")
+                .setSchema(defaultCatalogSchema)
+                .build();
+        Session hiveCatalogSession = testSessionBuilder()
+                .setCatalog(TRINO_HIVE_CATALOG)
+                .setSchema(defaultCatalogSchema)
+                .build();
+        assertUpdate(defaultCatalogSession, "CREATE TABLE tabledefault (data integer)");
+        assertUpdate(defaultCatalogSession, "INSERT INTO tabledefault VALUES (1),(2),(3),(4)", 4);
+
+        String customCatalogSchema = "custom_catalog_schema";
+        assertUpdate(
+                testSessionBuilder()
+                        .setCatalog(TRINO_HIVE_CUSTOM_CATALOG)
+                        .build(),
+                "CREATE SCHEMA " + customCatalogSchema);
+        Session customCatalogSession = testSessionBuilder()
+                .setCatalog(TRINO_HIVE_CUSTOM_CATALOG)
+                .setSchema(customCatalogSchema)
+                .build();
+        assertUpdate(customCatalogSession, "CREATE TABLE tablecustom (data integer)");
+        assertUpdate(customCatalogSession, "INSERT INTO tablecustom VALUES (4),(5),(6),(7)", 4);
+
+        // schemas from the default Hive catalog are not visible in the custom Hive catalog and vice versa
+        assertThat(query(defaultCatalogSession, "SHOW SCHEMAS")).matches("VALUES VARCHAR 'default', VARCHAR 'default_catalog_schema', VARCHAR 'information_schema'");
+        assertThat(query(hiveCatalogSession, "SHOW SCHEMAS")).matches("VALUES VARCHAR 'default', VARCHAR 'default_catalog_schema', VARCHAR 'information_schema'");
+        assertThat(query(customCatalogSession, "SHOW SCHEMAS")).matches("VALUES VARCHAR '" + customCatalogSchema + "', VARCHAR 'information_schema'");
```

Comment:
We could use `computeActual` instead for avoiding redundant VALUES, VARCHAR and '':
```java
        assertThat(computeActual(hiveCatalogSession, "SHOW SCHEMAS").getOnlyColumn())
                .containsOnly("default", "default_catalog_schema", "information_schema");
        assertThat(computeActual(customCatalogSession, "SHOW SCHEMAS").getOnlyColumn())
                .containsOnly(customCatalogSchema, "information_schema");
```